tests/__init__.py

```diff
@@ -551,5 +552,106 @@ def test_batch_lock_key(self):
+        self._ensure_queues(scheduled={'a': 1})
+
+        # Test canceling a scheduled task.
+        task.cancel()
+        self._ensure_queues()
+
+        # Canceling again raises an error
+        self.assertRaises(TaskNotFound, task.cancel)
+
+    def test_delay_scheduled_2(self):
+        task = Task(self.tiger, simple_task, queue='a')
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'a': 1})
+
+        task_id = task.id
+
+        # We can't look up a non-unique task by recreating it.
+        task = Task(self.tiger, simple_task, queue='a')
+        self.assertRaises(TaskNotFound, task.cancel)
+
+        # We can look up a task by its ID.
+        fetch_task = lambda: Task.from_id(self.tiger, 'a', 'scheduled', task_id)
+
+        task = fetch_task()
+        task.cancel()
+        self._ensure_queues()
+
+        # Task.from_id raises if it doesn't exist.
+        self.assertRaises(TaskNotFound, fetch_task)
+
+    def test_delay_scheduled_3(self):
+        task = Task(self.tiger, simple_task, unique=True)
+        task.delay(when=datetime.timedelta(minutes=5))
+        self._ensure_queues(scheduled={'default': 1})
+
+        # We can look up a non-unique task by recreating it.
+        task = Task(self.tiger, simple_task, unique=True)
+        task.cancel()
+        self._ensure_queues()
+
+    def test_execute(self):
+        task = Task(self.tiger, exception_task)
+        self.assertRaises(StandardError, task.execute)
+
+    def test_tasks_from_queue(self):
+        task0 = Task(self.tiger, simple_task)
+        task1 = Task(self.tiger, exception_task)
+        task2 = Task(self.tiger, simple_task, queue='other')
+
+        task0.delay()
+        task1.delay()
+        task2.delay()
+
+        n, tasks = Task.tasks_from_queue(self.tiger, 'default', 'queued')
+        self.assertEqual(n, 2)
+        self.assertEqual(task0.id, tasks[0].id)
+        self.assertEqual(task0.func, simple_task)
+        self.assertEqual(task0.func, tasks[0].func)
+        self.assertEqual(task0.serialized_func, 'tests.tasks.simple_task')
+        self.assertEqual(task0.serialized_func, tasks[0].serialized_func)
+        self.assertEqual(task0.state, tasks[0].state)
+        self.assertEqual(task0.state, 'queued')
+        self.assertEqual(task0.queue, tasks[0].queue)
+        self.assertEqual(task0.queue, 'default')
+
+    def test_eager(self):
+        self.tiger.config['ALWAYS_EAGER'] = True
+
+        # Ensure task is immediately executed.
+        task = Task(self.tiger, simple_task)
+        task.delay()
+        self._ensure_queues()
+
+        # Ensure task is immediately executed.
+        task = Task(self.tiger, exception_task)
+        self.assertRaises(StandardError, task.delay)
+        self._ensure_queues()
+
+        # Ensure there is an exception if we can't serialize the task.
+        task = Task(self.tiger, decorated_task,
+                    args=[datetime.datetime.utcnow()])
```

Comment:
Could add a comment why the task is not serializable in this example (I assume it's bcos datetime is not json-serializable by default, but it's just a hunch that's not clear from just looking at this code).